src/crypto/DeviceList.js

```diff
@@ -357,13 +373,24 @@ export default class DeviceList {
     /**
      * Find a device by curve25519 identity key
      *
-     * @param {string} userId     owner of the device
+     * @param {string} userId     owner of the device (optional)
      * @param {string} algorithm  encryption algorithm
      * @param {string} senderKey  curve25519 key to match
      *
      * @return {module:crypto/deviceinfo?}
      */
     getDeviceByIdentityKey(userId, algorithm, senderKey) {
+        if (arguments.length === 2) {
+            // if userId is omitted, shift the other arguments, and look up the
+            // userid
+            senderKey = algorithm;
+            algorithm = userId;
+            userId = this._userByIdentityKey[senderKey];
+            if (!userId) {
+                return null;
+            }
+        }
+
```

Comment:
This is a bit magic, but given the name of the function I can't really suggest anything better. Maybe in the future we can deprecate the 3 arg usage of this function altogether.